backend/src/org/commcare/session/SessionInstanceBuilder.java

```diff
@@ -39,7 +39,8 @@ private static void addSessionNavData(TreeElement sessionRoot, SessionFrame fram
      */
     private static void addDatums(TreeElement sessionData, SessionFrame frame) {
         for (StackFrameStep step : frame.getSteps()) {
-            if (SessionFrame.STATE_DATUM_VAL.equals(step.getType())) {
+            if (SessionFrame.STATE_DATUM_VAL.equals(step.getType()) ||
```

Comment:
Resolution to earlier: so previously every stack push was actually marked as a DATUM (Ie CASE_ID) type:

https://github.com/dimagi/commcare-core/blob/e91993ad92b3c445f95f48e31de63a6eee9daee5/backend/src/org/commcare/xml/StackFrameStepParser.java#L28-L28

So the logic here was still working. Now those are marked as UNKNOWN and later evaluated to COMPUTED types here.